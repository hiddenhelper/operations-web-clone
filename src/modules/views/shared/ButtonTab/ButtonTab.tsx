import React, { memo, useCallback } from 'react';

import Button from '@material-ui/core/Button';

import { getConditionalDefaultValue } from '../../../../utils/generalUtils';
import { tableGlobalStyles } from '../../../../assets/styles';

export interface IButtonTabProps {
  optFilter: { id: any; title: string; key: string };
  isActive: boolean;
  setFilter: (filter: any) => void;
}

const ButtonTab = ({ optFilter, isActive, setFilter }: IButtonTabProps) => {
  const tableGlobalClasses = tableGlobalStyles();
  const changeFilter = useCallback(() => setFilter(optFilter.id), [optFilter.id, setFilter]);

  return (
    <Button
      key={optFilter.id}
      data-testid="filter-status-opt"
      className={getConditionalDefaultValue(isActive, tableGlobalClasses.activeFilter, '')}
      onClick={changeFilter}
    >
      {optFilter.title}
    </Button>
  );
};

export default memo(ButtonTab);
