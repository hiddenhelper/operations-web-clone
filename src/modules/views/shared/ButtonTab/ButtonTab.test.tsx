import React from 'react';
import { render, act, fireEvent, RenderResult } from '@testing-library/react';

import ButtonTab, { IButtonTabProps } from './ButtonTab';

describe('ButtonTab', () => {
  let wrapper: RenderResult;
  let props: IButtonTabProps;

  beforeEach(() => {
    props = {
      optFilter: { id: 1, title: 'filter', key: 'filter-tab' },
      currentFilter: 'filter-tab',
      setFilter: jest.fn(),
    };
  });

  it('should render', () => {
    wrapper = render(<ButtonTab {...props} />);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should change filter', () => {
    wrapper = render(<ButtonTab {...props} />);

    act(() => {
      fireEvent.click(wrapper.getByTestId('filter-status-opt'));
    });

    expect(props.setFilter).toHaveBeenCalled();
  });
});
