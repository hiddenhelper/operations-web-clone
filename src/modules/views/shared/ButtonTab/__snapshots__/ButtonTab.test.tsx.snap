// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTab should render 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    data-testid="filter-status-opt"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      filter
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
